stacks-signer/src/runloop.rs

```diff
@@ -235,12 +236,33 @@ impl RunLoop {
                     }
                 }
             }
-            let new_signer = Signer::from(new_signer_config);
+            let mut new_signer = Signer::from(new_signer_config);
+
+            let dkg_id = retry_with_exponential_backoff(|| {
+                self.stacks_client
+                    .get_last_round(reward_cycle)
+                    .map_err(backoff::Error::transient)
+            })?
+            .unwrap_or(0);
+
+            let approved_aggregate_key = retry_with_exponential_backoff(|| {
+                self.stacks_client
+                    .get_approved_aggregate_key(reward_cycle)
+                    .map_err(backoff::Error::transient)
+            })?;
+
+            new_signer.state_machine.reset(dkg_id, &mut OsRng);
```

Comment:
Can you add an inline-comment here that OsRng is the same source of randomness used in `wsts::state_machine::signer::Signer::new()`?

Reply:
I can do that. However I think the plan is to remove the hard-coded `OsRng` in `wsts`. I created an issue for it https://github.com/Trust-Machines/wsts/issues/63, after Joey commented in my previous PR https://github.com/stacks-network/stacks-core/pull/4659#discussion_r1560665096

With that in mind, do you have any other preferences on how we manage RNGs here?

Reply:
Using `OsRng` seems fine to me, but maybe a safer strategy would be to add something like:

```
/// This is the RNG implementation that the signer uses when randomness is required for cryptographic operations. Currently, this is OsRng, which is also the RNG used by WSTS when initializing signer state.
pub type SignerRng = OsRng;
```

And use that throughout. Then, whenever that WSTS issue is cleared, it'll be easy just set the RNG during initialization.

Reply:
> Then, whenever that WSTS issue is cleared, it'll be easy just set the RNG during initialization.

I'm not sure I follow what you mean here. The `Signer::reset` method won't change in the WSTS issue since it is already generic. The problem is that WSTS hard-codes the rng in some other places that calls this function which it should not do as a library.

I don't see what benefit a type alias would give us here. If we want to refactor the code so that we can easily change RNG in the future a simple type alias doesn't cut it, because most RNGs would have an internal state so you'd like to pass it around in the code instead of creating it on the fly which we can do with OsRng.

A middle-ground would be to create a method to return the rng as

```
/// This is the RNG implementation that the signer uses when randomness is required for cryptographic operations. Currently, this is OsRng, which is also the RNG used by WSTS when initializing signer state.
const fn signer_rng() -> impl CryptoRng { OsRng }
```

This would make it easier to swap between OsRng and ThreadRng, and possibly some other types of Rngs.

Reply:
It's so that whenever `SignerState::new()` is changed to `SignerState::new(rng: R)`, it's very clear that it should be invoked in the signer binary using `SignerState::new(&mut SignerRng)`.

Reply:
Updated to use a single function as discussed yesterday ✅